spec/workers/deliver_submission_worker_spec.rb

```diff
@@ -40,29 +40,30 @@
   end
 
   describe 'locking' do
-    let(:submission) { create(:submission, state: Submission::QUEUED) }
-
-    before do
-      # Rewrite Submission#deliver to just perform a SQL update
-      allow(submission).to receive(:deliver) {
-        submission.update(reference: '1')
-      }
-    end
+    let(:submission) { create(:submission, :queued) }
 
     it 'calls Submission#deliver with SQL locking' do
-      # Update record in the DB, this doesn't update the in memory Ruby
-      # submission object
+      # Update all records in the DB; this doesn't update the in memory Ruby
+      # submission instance. This is simulating another worker running.
       Submission.update(state: Submission::DELIVERED)
-      # Prove submission object still has the old state:
+
+      # Prove this in memory submission instance still has the old state:
       expect(submission.state).to eq Submission::QUEUED
-      # Run `perform`, this causes a SQL update to set the reference - see the
-      # before block above
+
+      # Stub `Submission#deliver`as calling the original implementation isn't
+      # any good as it would try create a remote request which is obviously
+      # outside the concern for this test. The stubbed implementation ...
+      # Prove these methods are called:
+      expect(submission).to receive(:deliver)
+      expect(submission).to receive(:with_lock).and_call_original
+
+      # Run the worker job.
       perform
-      # Because we wrap the method call with `with_lock` the object is reloaded,
-      # without this the state would have remain as QUEUED. Prove submission is
-      # in the new state AND has the reference:
+
+      # Because of the `with_lock` block the submission instance gets reloaded.
+      # Without this the state would have remained as QUEUED and saved into the
+      # database. Prove submission is in the state from 1:
```

Comment:
> Prove submission is in the state _from 1_:

Don't quite understand this – I think you mean "Prove submission is in the state set by `Submission.update(state: Submission::DELIVERED)`", and that it can't have been updated by `perform` because we've stubbed the implementation.

Reply:
Ooops I had the comment blocks numbered, then I took out for some reason